bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusHandlerFactory.java

```diff
@@ -43,21 +50,72 @@
 
     private final Map<ThingUID, @Nullable ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
 
+    private @Nullable HttpService httpService;
+    private @Nullable StorageService storageService;
+    private @Nullable AccountsServlet accountsServlet;
+
     private static final Collection<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Arrays.asList(THING_TYPE_SENSEGUARD,
             THING_TYPE_SENSE, THING_TYPE_BRIDGE_ACCOUNT);
 
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.DYNAMIC)
+    protected void setHttpService(HttpService httpService) {
+        this.httpService = httpService;
+    }
+
+    protected void unsetHttpService(HttpService httpService) {
+        this.httpService = null;
+    }
+
+    @Reference
+    protected void setStorageService(StorageService storageService) {
+        this.storageService = storageService;
+    }
+
+    protected void unsetStorageService(StorageService storageService) {
+        this.storageService = null;
+    }
+
     @Override
     public boolean supportsThingType(ThingTypeUID thingTypeUID) {
         return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
     }
 
+    @Override
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+        HttpService httpService = this.httpService;
+        if (accountsServlet == null && httpService != null) {
+            accountsServlet = new AccountsServlet(httpService);
```

Comment:
Just in theory: Would it make sense to make `AccountsServlet` an OSGi service that references `HttpService`. And then you would reference `AccountsServlet` here in the factory and no longer `HttpService`.

Reply:
Agreed!

> I mean if we use something heavy like OSGi in the background we should use it, shouldn't we.

Totally agreeed! :D